packages/rpc-methods/src/restricted/notify.ts

```diff
@@ -9,11 +9,20 @@ import { ethErrors } from 'eth-rpc-errors';
 
 const methodName = 'snap_notify';
 
-// Move all the types to a shared place when implementing more notifications
-export enum NotificationType {
-  Native = 'native',
-  InApp = 'inApp',
-}
+// TODO: Move all the types to a shared place when implementing more
+// notifications.
+// Note: We can't use an enum here, because strings are not assignable to
```

Comment:
Hmm, this seems bad. Is this a compromise we have to make to get the request function typed? 🤔

Reply:
It's still typed, but using a union rather than an enum.

Reply:
Yeah, I get that. But enums are nice in general 😅 Maybe this is a trade off we should discuss with the team?

Reply:
You can create a union type from an enum to get benefits of both.


```typescript
export enum NotificationTypeEnum {
  Native = 'native',
  InApp = 'inApp',
}
export type NotificationType = keyof typeof NotificationTypeEnum;
```

https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-compile-time